Sources/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegate.swift

```diff
@@ -6,12 +6,25 @@
 
 import Foundation
 
+/// In case that you are not able to use `DDURLSessionDelegate` or its subclasses in your `URLSession` instance,
+/// you can use `DDURLSessionDelegateProviding` protocol.
+/// Make sure that your type implements the same `URLSessionTaskDelegate` methods as `DDURLSessionDelegate` does.
+/// Otherwise your `URLSession` will not be instrumented automatically.
+@objc
+public protocol DDURLSessionDelegateProviding {
+    var delegate: DDURLSessionDelegate { get }
+}
+
 /// The `URLSession` delegate object which enables network requests instrumentation. **It must be
 /// used together with** `Datadog.Configuration.trackURLSession(firstPartyHosts:)`.
 ///
 /// All requests made with the `URLSession` instrumented with this delegate will be intercepted by the SDK.
 @objc
-open class DDURLSessionDelegate: NSObject, URLSessionTaskDelegate {
+open class DDURLSessionDelegate: NSObject, URLSessionTaskDelegate, DDURLSessionDelegateProviding {
+    public var delegate: DDURLSessionDelegate {
+        return self
+    }
```

Comment:
It is IMO strange to see generic `delegate` property on the `delegate` object. Naming it `ddDelegate` / `ddForwardingDelegate` may sound better.